libs/cart/src/effects/cart-item.effects.ts

```diff
@@ -65,16 +72,33 @@ export class DaffCartItemEffects<
   add$ = this.actions$.pipe(
     ofType(DaffCartItemActionTypes.CartItemAddAction),
     switchMap((action: DaffCartItemAdd<U>) =>
-      this.driver.add(
-        this.storage.getCartId(),
-        action.input
-      ).pipe(
-        map((resp: V) => new DaffCartItemAddSuccess(resp)),
+      combineLatest([
+				this.store.pipe(select(
+					getDaffCartItemEntitiesSelectors<V, X, T>().selectAllCartItems)
+				),
+				this.driver.add(this.storage.getCartId(), action.input)
+			]
+			).pipe(
+        map(([cartItems, resp]: [T[], V]) => {
+					//find the cart item that is new or that has a different qty and change the state to "new"
+					//I think we also need to attach the cart item id to the action, so the "new" state can be 
+					//changed back to "default" in a "reset" effect below.
```

Comment:
Seems good, how would you find the cart ID that has changed?

Reply:
I would check if the length of the cartItems array has changed. If it has, find the cart item id that doesn't belong (it would be nice if we could just take the last cart item, but I'm not sure the new cart item will always be the last one). If the length of the array hasn't changed, compare the quantities of cart items and take the cart item with a different quantity.

Reply:
I think:

> new cart item will always be the last

will always hold.

>compare the quantities of cart items and take the cart item with a different quantity.

This is a good solve!

Reply:
@damienwebdev ok I'll go with taking the last item when possible then

Reply:
> I think:
> 
> > new cart item will always be the last
> 
> will always hold.

Isn't that platform specific?

Reply:
It is. In retrospect, I think we actually don't care about which one was added, we care about showing the user the updates to the items in their order, so we'd essentially be diffing ANY changes to the quantities and showing notices for those items.

Reply:
So if it's platform specific, I should not do it, right? I should just be checking if it is there or if the quantity changed

Reply:
I think we should be diffing the item quantities; it's the safer and more robust solution.